symfony/client/src/orangehrmRecruitmentPlugin/pages/EditJobVacancy.vue

```diff
@@ -0,0 +1,253 @@
+            />
+          </div>
+        </oxd-grid>
+        <br />
+        <oxd-divider />
+        <oxd-form-actions>
+          <required-text />
+          <oxd-button displayType="ghost" label="Cancel" @click="onCancel" />
+          <submit-button />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+<script>
+import {APIService} from '@/core/util/services/api.service';
+import {navigate} from '@orangehrm/core/util/helper/navigation';
+import SwitchInput from '@orangehrm/oxd/core/components/Input/SwitchInput';
+
+import {
+  required,
+  shouldNotExceedCharLength,
+  digitsOnly,
+  max,
+} from '@orangehrm/core/util/validation/rules';
+import EmployeeAutocomplete from '@/core/components/inputs/EmployeeAutocomplete';
+import JobtitleDropdown from '@/orangehrmPimPlugin/components/JobtitleDropdown';
+import VacancyLinkCard from '../components/VacancyLinkCard.vue';
+
+const vacancyModel = {
+  jobTitle: null,
+  name: '',
+  hiringManager: null,
+  numOfPositions: '',
+  description: '',
+  status: false,
+  isPublished: false,
+};
+
+export default {
+  props: {
+    vacancyId: {
+      type: String,
+      required: true,
+    },
+  },
+  name: 'edit-job-vacancy',
+  components: {
+    'oxd-switch-input': SwitchInput,
+    'employee-autocomplete': EmployeeAutocomplete,
+    'jobtitle-dropdown': JobtitleDropdown,
+    'vacancy-link-card': VacancyLinkCard,
+  },
+
+  data() {
+    return {
+      isLoading: false,
+      rssFeedUrl: '',
+      webFeedUrl: '',
+      currentName: '',
+      vacancy: {...vacancyModel},
+      rules: {
+        jobTitle: [required],
+        name: [required, shouldNotExceedCharLength(50)],
+        hiringManager: [required],
+        numOfPositions: [max(99), digitsOnly],
+        description: [],
+        status: [required],
+        isPublished: [required],
+      },
+    };
+  },
+  setup() {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      'api/v2/recruitment/vacancies',
+    );
+    return {
+      http,
+    };
+  },
+
+  methods: {
+    onCancel() {
+      navigate('/recruitment/viewJobVacancy');
+    },
+    onSave() {
+      this.isLoading = true;
+      this.vacancy = {
+        name: this.vacancy.name,
+        jobTitleId: this.vacancy.jobTitle.id,
+        employeeId: this.vacancy.hiringManager.id,
+        numOfPositions: this.vacancy.numOfPositions,
+        description: this.vacancy.description,
+        status: this.vacancy.status ? 1 : 0,
+        isPublished: this.vacancy.isPublished ? 1 : 0,
+      };
+      this.http
+        .update(this.vacancyId, {...this.vacancy})
+        .then(() => {
+          return this.$toast.saveSuccess();
+        })
+        .then(() => {
+          this.onCancel();
+        });
+    },
+    getUrls() {
+      const baseUrl = window.location.hostname;
+      const urlSplits = baseUrl.split('/');
+      console.log(urlSplits);
+    },
+  },
+  created() {
+    this.isLoading = true;
+    this.getUrls();
```

Comment:
hope this method is only for dev purposes?

Reply:
yeah. Removed it